crates/kitsune_p2p/kitsune_p2p/src/lib.rs

```diff
@@ -26,3 +26,14 @@ pub mod test_util;
 mod test;
 
 pub mod fixt;
+
+/// 10MB of entropy free for the taking.
+/// Useful for initializing arbitrary::Unstructured data
+#[cfg(test)]
+pub static NOISE: once_cell::sync::Lazy<Vec<u8>> = once_cell::sync::Lazy::new(|| {
```

Comment:
this is going to be in-memory... do we need it to be repeatable or something? wouldn't it be better to generate how much we need as it is needed?

Reply:
I tried thinking away around this but it looks like the Unstructured type takes a slice and there's no way for it to demand new unstructured data as needed. I really don't know enough about this to say either way. @maackle might have to comment. Maybe we can merge this and change it in a follow up once he is back?

Reply:
Yes I wish it were possible to generate data on the fly too, but like Tom said `arbitrary::Unstructured` takes a slice, so the entropy needs to be preallocated. It makes more sense in the context of property testing, where the unstructured data gets gradually altered over successive test runs, which is I think what `arbitrary` was mainly designed for. I think it's OK that we use the same `NOISE` for every test, and I was thinking 10MB is an acceptable amount to keep in memory.

Reply:
Cool :+1: